### PR TITLE
Staging Review Accessibility Fixes for 10-10CG POA work

### PR DESCRIPTION
**Pull request by LinnJS**
## Description
These are fixes that were founding during VSP Collaboration Cycle Review and are documented in this [Ticket](https://app.zenhub.com/workspaces/vsa---caregiver-5fff0cfd1462b6000e320fc7/issues/department-of-veterans-affairs/va.gov-team/23616)

## Testing done
- Manual testing

## Acceptance criteria
- [x] [Dev intent] Ensure `legend`s are properly used with `fieldset`s
- [x] Style the `legend` to be more recognizable from standard `p` text [Step 4 of 5: Legal Representative](https://preview.uxpin.com/3bf6496017f55041a94c2cfc8009c35dad5a79f2#/pages/137666495/simulate/sitemap?mode=i) (might just be a prototype issue)
- [x] Add labels to signature boxes to satisfy accessibility requirements
- [ ] If we keep "on behalf of" - Either rewrite content so that "on behalf of" is not an isolated `p` tag between two fields (preferred) OR use `aria-describedby` so that it is programatically linked to the first field [Step 5 of 5: Review Application](https://preview.uxpin.com/3bf6496017f55041a94c2cfc8009c35dad5a79f2#/pages/137666520/simulate/sitemap?mode=i)
- [ ] Make document upload edit by default on review page 

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by craigwheeler — APPROVED** [review]

File: src/applications/caregivers/components/PreSubmitInfo/SignatureCheckbox.jsx [hunk 1]
```diff
@@ -49,7 +50,10 @@ const SignatureCheckbox = ({
         />
 
         {isRepresentative && (
-          <p className="on-behalf-representative">
+          <p
+            className="on-behalf-representative"
+            id={`${label}-signature-label`}
+          >
```
Comment (LinnJS) [inline src/applications/caregivers/components/PreSubmitInfo/SignatureCheckbox.jsx:56]: I implemented this code but will not work till the feature I noted in the ticket is completed on the DSVA global component

File: src/applications/caregivers/config/chapters/signAsRepresentative/uploadPOADocument.js [hunk 2]
```diff
@@ -36,39 +36,34 @@ export default {
   uiSchema: {
     'ui:description': RepresentativeDocumentUploadDescription(),
     'ui:title': () => (
-      <legend className="vads-u-font-size--h4">
+      <h3 className="vads-u-font-size--h4 vads-u-margin--0">
```
Comment (LinnJS) [inline src/applications/caregivers/config/chapters/signAsRepresentative/uploadPOADocument.js:39]: Removed legend in favor of new heading level

File: src/applications/caregivers/sass/signatureBox.scss [hunk 3]
```diff
@@ -37,7 +37,7 @@
       margin: 10px 0 0 0;
     }
 
-    section {
+    .wide-input {
```
Comment (LinnJS) [inline src/applications/caregivers/sass/signatureBox.scss:40]: Fixed styling bug

File: src/applications/caregivers/config/form.js [hunk 4]
```diff
@@ -227,4 +228,7 @@ const formConfig = {
   },
 };
 
+/* TODO Need to change editModeOnReviewPage for document upload to true
```
Comment (LinnJS) [inline src/applications/caregivers/config/form.js:231]: Could not fix document upload review page request due to VSP bug reference ticket for details.

File: src/applications/caregivers/components/PreSubmitInfo/SignatureInput.jsx [hunk 5]
```diff
@@ -42,6 +39,10 @@ const SignatureInput = ({
   const getName = (middle = '') =>
     removeSpaces(`${firstName}${middle}${lastName}`);
 
+  const errorMessage = isRepresentative
+    ? 'You must sign as representative.'
+    : `Your signature must match previously entered name: ${getName()}`;
```
Comment (LinnJS) [inline src/applications/caregivers/components/PreSubmitInfo/SignatureInput.jsx:44]: Added full name to signature box error message

**Comment by LinnJS** [comment]
Just a note I tried to get the last two fixes into this PR but there was a global bug on the document upload solution and a feature that does not exist on the DSVA component that needs to be implemented so out of my control at this moment to get those fixes in this PR.

**Pull request merged**